### PR TITLE
[circle-eval-diff] Introduce CircleEvalDiff class

### PR DESCRIPTION
**Pull request by jinevening**
This introduces the CircleEvalDiff class.

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Related to: https://github.com/Samsung/ONE/issues/8634
Draft PR: https://github.com/Samsung/ONE/pull/8741

### CODE REVIEW

File: compiler/circle-eval-diff/src/CircleEvalDiff.h [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+namespace circle_eval_diff
+{
+
+// Forward declaration
+class ModuleEvalDiff;
+
+enum class Metric
+{
+  Invalid, // For debugging
```
Comment (mhs4670go) [inline compiler/circle-eval-diff/src/CircleEvalDiff.h:33]: (optional) How about use `undefined` for debugging and initialization?

File: compiler/circle-eval-diff/src/CircleEvalDiff.h [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  MAE,
+};
+
+enum class InputFormat
+{
+  Invalid, // For debugging
```
Comment (mhs4670go) [inline compiler/circle-eval-diff/src/CircleEvalDiff.h:39]: ditto

File: compiler/circle-eval-diff/src/CircleEvalDiff.h [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  H5,
+  // TODO Implement Random, Directory
+};
+
+class CircleEvalDiff final
+{
+public:
+  struct Context
+  {
+    std::string first_model_path;
```
Comment (mhs4670go) [inline compiler/circle-eval-diff/src/CircleEvalDiff.h:49]: Adding `<string>` header would be better.

File: compiler/circle-eval-diff/src/CircleEvalDiff.cpp [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+std::unique_ptr<luci::Module> import(const std::string &model_path)
+{
+  // Load model from the file
+  std::ifstream fs(model_path, std::ifstream::binary);
+  if (fs.fail())
+  {
+    throw std::runtime_error("Cannot open model file \"" + model_path + "\".\n");
+  }
+  std::vector<char> model_data((std::istreambuf_iterator<char>(fs)),
+                               std::istreambuf_iterator<char>());
```
Comment (mhs4670go) [inline compiler/circle-eval-diff/src/CircleEvalDiff.cpp:38]: (optional) `foder` library can be applied.

**Review by mhs4670go — DISMISSED** [review]
LGTM

File: compiler/circle-eval-diff/src/CircleEvalDiff.cpp (hunk 4 above)
Comment (seanshpark) [reply compiler/circle-eval-diff/src/CircleEvalDiff.cpp:38]: @jinevening , if you are to revise using `foder`, should I wait for that change or maybe revise in another PR?

**Review by seanshpark — DISMISSED** [review]
LGTM

File: compiler/circle-eval-diff/src/CircleEvalDiff.cpp (hunk 4 above)
Comment (jinevening) [reply compiler/circle-eval-diff/src/CircleEvalDiff.cpp]: I've updated this PR. PTAL :)

**Review by mhs4670go — APPROVED** [review]
LGTM. Thank you:)

**Review by seanshpark — APPROVED** [review]
LGTM

**Pull request merged**